src/pattern-library/components/patterns/data/DataTablePage.js

```diff
@@ -0,0 +1,440 @@
+              to values.
+            </p>
+            <p>
+              Only fields that are present in the <code>columns</code> data will
+              be rendered. In this example, the <code>translatedTitle</code>{' '}
+              field is not referenced in <code>columns</code>, so it is not
+              rendered. In other words, row objects may contain entries that are
+              not used by <code>DataTable</code>.
+            </p>
+            <Next.Code
+              title="columns data used for this example"
+              content={`const columns = [
+  { field: 'title', label: 'Title' },
+  { field: 'year', label: 'Year' },
+  { field: 'language', label: 'Language' },
+];`}
+            />
+            <Library.Demo withSource>
+              <div className="w-full h-[250px]">
+                <Scroll>
+                  <DataTable
+                    title="A subset of Nabokov's novels with publish date and original language"
+                    rows={nabokovRows}
+                    columns={nabokovColumns}
+                  />
+                </Scroll>
+              </div>
+            </Library.Demo>
+          </Library.Example>
+
+          <Library.Example title="columns">
+            <p>
+              Columns define the headers that are rendered, as well as which row
+              fields are rendered.
+            </p>
+            <p>
+              <code>field</code> and <code>label</code> are required. Optional{' '}
+              <code>classes</code> will extend CSS classes applied to associated{' '}
+              <code>th</code> elements and is mainly intended for setting the
+              width of columns.
+            </p>
+            <Next.Code
+              title="columns data used for this example"
+              content={`const columns = [
+  { field: 'title', label: 'Title', classes: 'w-[80%]' },
+  { field: 'year', label: 'Year' },
+];`}
+            />
+            <Library.Demo withSource>
+              <div className="w-full h-[250px]">
+                <Scroll>
+                  <DataTable
+                    title="Some of Nabokov's novels and their publish date"
+                    rows={nabokovRows}
+                    columns={[
+                      { field: 'title', label: 'Title', classes: 'w-[80%]' },
+                      { field: 'year', label: 'Year' },
+                    ]}
+                  />
+                </Scroll>
+              </div>
+            </Library.Demo>
+          </Library.Example>
+
+          <Library.Example title="renderItem">
+            <p>
+              <code>renderItem</code> is a callback for formatting the contents
+              of an individual table cell. It is given the current{' '}
+              <code>row</code> and the <code>field</code> to format. Returning{' '}
+              <code>null</code> will apply default formatting for the given{' '}
+              <code>field</code> value.
```

Comment:
This might cause some confusion if a caller wanted to render nothing for a particular cell (since returning `null` is the normal way for a rendering function to generate no output). Since the default formatting is just `row[field]`, we could move the responsibility for doing that to the `renderItem` callback if provided. In other words, if the caller provides a `renderItem` callback, it must handle all fields. Otherwise the default formatting is used for all fields. An alternative would be to specify/document that callers should return an empty string or other non-nullish but empty value in this case.

Reply:
Yeah, you're right. This has had a minor code smell the whole time. Let's assert that a `renderItem` callback needs to handle all fields.